src/AvroConvert/AvroObjectServices/BuildSchema/ReflectionSchemaBuilder.cs

```diff
@@ -479,18 +486,9 @@ public List<RecordFieldSchema> BuildRecordFields(
                             $"Type member '{info.MemberInfo.GetType().Name}' is not supported."));
                 }
 
-                TypeSchema fieldSchema;
-
-                if (info.AvroTypeRepresentation.HasValue)
```

Comment:
This logic has been pushed down lower in the resolution tree, at the primitive/logical type level. This allows for unions to be created to support null types.